### ISSUE TITLE
[

### ISSUE THREAD

**Issue by shixiedesign** [carbon-design-system/carbon#3849]
@tw15egan @shixiedesign  This came up in one of our Slack channels so I wanted to share what I posted there + some additional accessibility info I found...

## Accessibility

https://webaim.org/techniques/hypertext/

> Links must have a non-empty href attribute in order to be considered true links and to be accessible to keyboard users.

WebAIM doesn't even like ~`onClick`~ `onmouseover` when used with an `href` -- it considers it a link that goes nowhere for a keyboard user:

```
Bad Example
The link in this example goes nowhere. Its only purpose is to activate a JavaScript function.

<a href="#" onmouseover="dropdownmenu()">Products</a>
```

## Spec concerns

https://html.spec.whatwg.org/#the-a-element

From the spec above:

> If the `a` element has an `href` attribute, then it represents a hyperlink (a hypertext anchor) labeled by its contents.

> If the `a` element has no `href` attribute, then the element represents a placeholder for where a link might otherwise have been placed, if it had been relevant, consisting of just the element's contents.

So fundamentally the `href` attribute's presence = a link. Without an `href`, an anchor is a placeholder.
## Suggestion 

As TJ said, I think in this case the best solution is to advocate that someone use a `Button` with an `onClick` instead. A ghost type `Button` would even give a link-like appearance. 

And the benefit of going with a `Button` here is that you would be following general button vs anchor roles (i.e., anchors need to go somewhere and buttons need to be actions) and also keep things compliant with accessible tooling and general spec guidance.

Plus just wanted to point out that generally I like that `&:not([href]) { ... }` style block because it encourages conversations like this, which help us reconsider which HTML elements are right for the job 😅

_Originally posted by @jendowns in https://github.com/carbon-design-system/carbon/issues/3711#issuecomment-522591274_
